src/teams.rs

```diff
@@ -36,9 +40,14 @@ struct Data {
 const ENCODING_SET: AsciiSet = NON_ALPHANUMERIC.remove(b'-').remove(b'_');
 
 impl Data {
-    async fn load() -> Result<Self, Box<dyn Error>> {
+    async fn load(
+        teams_cache: &Arc<RwLock<RustTeams>>,
+    ) -> Result<Self, Box<dyn Error + Send + Sync>> {
         Ok(Data {
-            teams: crate::cache::get::<Vec<Team>>(get_teams).await?,
+            teams: RustTeams::get(teams_cache)
+                .await
+                .0
+                .ok_or_else(|| Box::<dyn Error + Send + Sync>::from("failed to load teams"))?,
```

Comment:
The teams data does currently cause an error if it cannot be fetched (unlike rust version and release post) so I used an option in the cache type to indicate there might not be any teams data.